src/java/org/apache/cassandra/locator/AbstractReplicationStrategy.java

```
@@ -324,7 +324,8 @@ private static AbstractReplicationStrategy createInternal(String keyspaceName,
         try
         {
             Constructor<? extends AbstractReplicationStrategy> constructor = strategyClass.getConstructor(parameterTypes);
-            strategy = constructor.newInstance(keyspaceName, tokenMetadata, snitch, strategyOptions);
+            IEndpointSnitch endpointSnitch = snitch == null ? new SimpleSnitch() : snitch;
```

This will propagate `null` snitch in client mode. I just have to put there something non-nullish.

---

to be extra safe, we could probably wrap this in DatabaseDescriptor.isClientOrToolInitialized() to set it to SimpleSnitch only in that case, otherwise we would just let this fail ...

---

Agreed. I did this in https://github.com/apache/cassandra/commit/fecac728ec945e9b376a5fb2c4af54f337d744cf.